server/bin/pbench-audit-server.py

```diff
@@ -0,0 +1,596 @@
+        prefix_path = Path(prefix)
+        if not prefix_path.name.startswith("prefix.") and not prefix_path.name.endswith(
+            ".prefix"
+        ):
+            hier.add_unexpected_controllers(
+                Hierarchy.NON_PREFIXES, controller, prefix_path.name
+            )
+        elif prefix_path.name.startswith("prefix."):
+            hier.add_unexpected_controllers(
+                Hierarchy.WRONG_PREFIXES, controller, prefix_path.name
+            )
+
+    return 0
+
+
+# 'hier' is an object of ArchiveHierarchy class imported from hierarchy.py
+def verify_archive(hier):
+
+    controllers = glob.iglob(os.path.join(hier.path, "*"))
+    verify_valid_controllers(hier, controllers)
+
+    # now check each "good" controller and get the tarballs it contains
+    for controller in hier.controllers:
+        direct_entries = glob.iglob(os.path.join(hier.path, controller, "*"))
+        hidden_entries = glob.glob(os.path.join(hier.path, controller, ".*"))
+        if hidden_entries:
+            for hid_entries in hidden_entries:
+                hid_entries_path = Path(hid_entries)
+                if hid_entries_path.is_file():
+                    hier.add_unexpected_controllers(
+                        Hierarchy.UNEXPECTED_OBJECTS, controller, hid_entries_path.name,
+                    )
+        controller_subdir = list()
+        for item in direct_entries:
+            item_path = Path(item)
+            if item_path.is_dir():
+                controller_subdir.append(item_path.name)
+            elif item_path.is_symlink():
+                try:
+                    item_p = item_path.resolve(strict=True)
+                except FileNotFoundError:
+                    print(
+                        f"{item_path.name}_path, '{item_path}', does not resolve to a real location"
+                    )
+                    return 1
+                symlink_item = f"{item_path.name} -> {item_p}"
+                hier.add_unexpected_controllers(
+                    Hierarchy.UNEXPECTED_SYMLINKS, controller, symlink_item
+                )
+            elif all(
+                [
+                    item_path.is_file(),
+                    not item_path.name.endswith(".tar.xz"),
+                    not item_path.name.endswith(".tar.xz.md5"),
+                ]
+            ):
+                hier.add_unexpected_controllers(
+                    Hierarchy.UNEXPECTED_OBJECTS, controller, item_path.name
+                )
+            elif item_path.is_file() and (
+                item_path.name.endswith(".tar.xz")
+                or item_path.name.endswith(".tar.xz.md5")
+            ):
+                hier.add_tarballs(controller)
+            else:
+                print(
+                    f"{item_path} item should have been handled by the above mentioned conditions. "
+                    f"It is an unexpected item which should not have occured, "
+                    f"leading to an inappropriate condition"
```

Comment:
Minor, but this message seems a little vague, can we specify what conditions the object did not meet so when someone looks at the log, he/she doesnt have to go to code to understand what are the `above mentioned conditions` (PS, I understand this might be a rare condition to occur)